src/main/java/core/basesyntax/service/impl/WriterImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.service.Writer;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Map;
+
+public class WriterImpl implements Writer {
+    private static final String REPORT_HEAD_LINE = "fruit,quantity";
+    private static final char COMMA = ',';
+
+    @Override
+    public void createReport(String reportPath) {
+        File report = new File(reportPath);
```

Comment:
Consider verifying whether the report file already exists and is writable before attempting to write to it. This can prevent potential issues where the file is not accessible or cannot be created.